scripts/solve_network.py

```diff
@@ -814,6 +814,36 @@ def add_pipe_retrofit_constraint(n):
     n.model.add_constraints(lhs == rhs, name="Link-pipe_retrofit")
 
 
+def add_ocgt_retrofit_constraint(n):
+    """
+    Add constraint for retrofitting existing OCGT to H2 OCGT.
+    """
+    # existing OCGT plants
+    gas_i = n.links.query(
+        "carrier == 'OCGT' and p_nom_extendable and p_nom > 1e-3"
+    ).index
+    h2_i = n.links.query("carrier == 'OCGT H2 retrofitted' and p_nom_extendable").index
+    if h2_i.empty or gas_i.empty:
+        return
+
+    # filter gas plants from current planning horizon
+    current_horizon = snakemake.wildcards.planning_horizons
+    gas_i = gas_i[~gas_i.str.endswith(current_horizon)]
+
+    # store p_nom value for rhs of constraint
+    p_nom = n.model["Link-p_nom"]
+
+    # sum of p_nom OCGT and retrofitted must be <= installed capacity of OCGT
+    CH4_per_H2 = (
+        n.links.loc[h2_i].efficiency.values / n.links.loc[gas_i].efficiency.values
+    )
+
+    lhs = CH4_per_H2 * (p_nom.loc[h2_i] + p_nom.loc[gas_i])
```

Comment:
Do you really need to take account for the efficiencies here? You will just have a lower electricity output if you retrofit but the p_nom which refers to maximum input at bus0 should stay the same